standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -499,6 +499,34 @@ public static void populateQuickStats(List<FileStatus> fileStatus, Map<String, S
     params.put(StatsSetupConst.NUM_ERASURE_CODED_FILES, Integer.toString(numErasureCodedFiles));
   }
 
+  public static void populateQuickStatsWithPrevStats(List<FileStatus> fileStatus, Map<String,String> params){
```

Comment:
this method has exactly the same logic as populateQuickStats except that in this case, initial integer values are set from parameters
we should find a way to have the same logic written once: iteration on FileStatus + setting integers back to parameters, otherwise there is a chance that the two methods will diverge